sspl_test/alerts/self_hw/test_node_bmc.py

```diff
@@ -0,0 +1,169 @@
+    channel_interface = Conf.get(
+        SSPL_CONF, "BMC_INTERFACE>default", 'system')
+
+    if channel_interface == "system":
+        # Check BMC is accessible through KCS
+        cmd = "sudo ipmitool channel info"
+        expected_channel = "KCS"
+        channel_found = None
+        res_op, res_err, res_rc = SimpleProcess(cmd).run()
+        if res_rc == 0:
+            res_op = res_op.decode()
+            search_res = re.search(
+                r"%s[\s]+:[\s]+(\w+)(.*)" % CHANNEL_PROTOCOL, res_op)
+            if search_res:
+                channel_found = search_res.groups()[0]
+            if expected_channel != channel_found:
+                print("UNEXPECTED BMC CHANNEL TYPE FOUND.")
+                print("Expected: %s" % expected_channel)
+                print("Found: %s" % channel_found)
+        else:
+            res_err = res_err.decode()
+            kcs_errors = ("could not find inband device", "driver timeout")
+            if not any(err for err in kcs_errors if err in res_err):
+                raise Exception("ERROR: %s" % res_err)
+    elif channel_interface == "lan":
+        # Check BMC is accessible through LAN
+        subcommand = "channel info"
+        bmc_ip = Conf.get(GLOBAL_CONF, BMC_IP_KEY)
+        bmc_user = Conf.get(GLOBAL_CONF, BMC_USER_KEY)
+        bmc_secret = Conf.get(GLOBAL_CONF, BMC_SECRET_KEY)
+        bmc_key = Cipher.generate_key(MACHINE_ID, "server_node")
+        bmc_passwd = Cipher.decrypt(
+            bmc_key, bmc_secret.encode("utf-8")).decode("utf-8")
+        cmd = "sudo ipmitool -H %s -U %s -P %s -I lan %s" %(
+            bmc_ip, bmc_user, bmc_passwd, subcommand)
+        res_op, res_err, res_rc = SimpleProcess(cmd).run()
+        if res_rc != 0:
+            raise Exception("ERROR: %s" % res_err.decode())
+
+def test_chassis_selftest(args):
+    """Check chassis selttestsel is passed."""
+    cmd = "ipmitool chassis selftest"
+    expected_res = "Self Test Results    : passed"
+    res_op, res_err, res_rc = SimpleProcess(cmd).run()
+    if res_rc == 0:
+        res_op = res_op.decode()
+        if expected_res not in res_op:
+            assert False, res_op
+    else:
+        raise Exception("ERROR: %s" % res_err.decode())
+
+def test_sensor_availability(args):
+    """Fail if any expected sensor is not detected by ipmitool."""
+    found_all_sensors = True
+    sensors = [
+        "Voltage",
+        "Temperature",
+        "Power Supply",
+        "Drive Slot / Bay",
+        "Fan"
+        ]
+    # Get manufacturer name
+    manufacturer = get_manufacturer_name()
+    for sensor in sensors:
+        cmd = ["ipmitool", "sdr", "type", sensor]
+        res_op, res_err, res_rc = SimpleProcess(cmd).run()
+        if res_rc == 0:
+            res_op = res_op.decode().replace("\n", "")
+            if not res_op:
+                found_all_sensors = False
+                print(
+                    "'%s' sensor is not seen in %s node server." % (
```

Comment:
Rename to get_server_details(0 and fetch more info.

Need more Server details here, below should suffice to identify the server to good extent
Board Mfg
Board Product
Board Part Number
Product Name
Product Part Number

Fru print command on different servers looks like below, look for FRU Device ID 0
> ipmitool fru print
FRU Device Description : Builtin FRU Device (ID 0)
 Chassis Type          : Other
 Chassis Part Number   : CSE-819UTS-R751P-T
 Chassis Serial        : C819UAI31D90101
 Board Mfg Date        : Thu Aug 22 21:30:00 2019
 Board Mfg             : **Supermicro**
 Board Product         : X11DPU
 Board Serial          : OM198S043534
 Board Part Number     : X11DPU
 Product Manufacturer  : Supermicro
 Product Part Number   : SYS-6019U-TRT
 Product Serial        : A264223X9B01906

> ipmitool fru print
FRU Device Description : Builtin FRU Device (ID 0)
 Board Mfg Date        : Wed Aug  5 21:21:00 2020
 Board Mfg             : **DELL**
 Board Product         : PowerEdge R640
 Board Serial          : CNCMS0008500ME
 Board Part Number     : 0H28RRA00
 Product Manufacturer  : DELL
 Product Name          : PowerEdge R640
 Product Version       : 01
 Product Serial        : 486WH63
 Product Asset Tag     :

> ipmitool fru print
FRU Device Description : Builtin FRU Device (ID 0)
 Chassis Type          : Rack Mount Chassis
 Chassis Part Number   : ..................
 Chassis Serial        : ..................
 Chassis Extra         : ...............................
 Chassis Extra         : ...............................
 Board Mfg Date        : Sun Sep  8 04:31:00 2019
 Board Mfg             : **Intel Corporation**
 Board Product         : S2600WFT
 Board Serial          : BQWF93500648
 Board Part Number     : H48104-872
 Product Manufacturer  : Intel Corporation
 Product Name          : S2600WFT
 Product Part Number   : K75470-001
 Product Version       : LWF1208YR814606
 Product Serial        : LUF993900186
 Product Asset Tag     : ....................

Reply:
Done.
Addressed in PR #402